x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityBwcRestIT.java

```diff
@@ -0,0 +1,245 @@
+
+    private static final Version OLD_CLUSTER_VERSION = Version.fromString(System.getProperty("tests.old_cluster_version"));
+
+    static {
+        fulfillingCluster = ElasticsearchCluster.local()
+            .version(OLD_CLUSTER_VERSION)
+            .distribution(DistributionType.INTEG_TEST)
+            .name("fulfilling-cluster")
+            .apply(commonClusterConfig)
+            .setting("xpack.ml.enabled", "false")
+            .build();
+
+        queryCluster = ElasticsearchCluster.local()
+            .version(Version.CURRENT)
+            .distribution(DistributionType.INTEG_TEST)
+            .name("query-cluster")
+            .apply(commonClusterConfig)
+            .setting("xpack.security.remote_cluster_client.ssl.enabled", "true")
+            .setting("xpack.security.remote_cluster_client.ssl.certificate_authorities", "remote-cluster-ca.crt")
+            .rolesFile(Resource.fromClasspath("roles.yml"))
+            .build();
+    }
+
+    @ClassRule
+    // Use a RuleChain to ensure that fulfilling cluster is started before query cluster
+    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
+
+    public void testBwcWithLegacyCrossClusterSearch() throws Exception {
+        final boolean useProxyMode = randomBoolean();
+        // Update remote cluster settings on QC.
+        setupQueryClusterRemoteClusters(useProxyMode);
+        // Ensure remote cluster is connected
+        ensureRemoteFulfillingClusterIsConnected(useProxyMode);
+
+        // Fulfilling cluster
+        {
+            // Index some documents, so we can attempt to search them from the querying cluster
+            final Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                { "index": { "_index": "index1" } }
+                { "foo": "bar" }
+                { "index": { "_index": "index2" } }
+                { "bar": "foo" }
+                { "index": { "_index": "prefixed_index" } }
+                { "baz": "fee" }\n"""));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+        }
+
+        // Query cluster
+        {
+            // Index some documents, to use them in a mixed-cluster search
+            final var indexDocRequest = new Request("POST", "/local_index/_doc?refresh=true");
+            indexDocRequest.setJsonEntity("{\"local_foo\": \"local_bar\"}");
+            assertOK(client().performRequest(indexDocRequest));
+
+            // Create user role with privileges for remote and local indices
+            final var putRoleRequest = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+            putRoleRequest.setJsonEntity("""
+                {
+                  "cluster": ["manage_own_api_key"],
+                  "indices": [
+                    {
+                      "names": ["local_index", "index1", "not_found_index"],
```

Comment:
Nit: for this test suite, I think we can drop both `not_found_index` and `prefixed_index` since those aren't not involved in tests. Since we only have local_index and index1 in that case, could even got for renaming index1 to remote_index, for clarity.